app/src/main/java/info/nightscout/androidaps/utils/stats/TddCalculator.kt

```diff
@@ -111,7 +82,9 @@ class TddCalculator @Inject constructor(
     }
 
     fun calculate(startTime: Long, endTime: Long): TotalDailyDose {
-        val tdd = TotalDailyDose(timestamp = startTime)
+        val startTimeAligned = startTime - startTime % (5 * 60 * 1000)
```

Comment:
this should be already alligned because startTime is midnight (well not XXXXX000 in milliseconds but the same timestamp all the time)

Reply:
it is aligned in case of daily tddCalc. but in [other cases ](https://github.com/justmara/AndroidAPS/blob/fix/tdd-cache/app/src/main/java/info/nightscout/androidaps/utils/stats/TddCalculator.kt#L78) `now()` is used as base. thats why 1 second alignment were here basically. but this 1 second mark misses badly, and we cant force callers to care about time alignment to match our internal caches. thats why interval is aligned here.